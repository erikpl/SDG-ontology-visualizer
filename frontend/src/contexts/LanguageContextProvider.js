import React, { useContext, useState } from 'react';
import LanguageContext from './LanguageContext';
/* eslint-disable react/prop-types */

export const useLanguageContext = () => useContext(LanguageContext);

export default function LanguageContextProvider({ children }) {
  const [language, changeLanguage] = useState('en');
  return (
    <LanguageContext.Provider value={{ language, changeLanguage }}>
      {children}
    </LanguageContext.Provider>
  );
}