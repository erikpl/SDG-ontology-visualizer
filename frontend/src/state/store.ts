import { combineReducers, createStore } from 'redux';
import apiErrorReducer from './reducers/apiErrorReducer';
import ontologyReducer from './reducers/ontologyReducer';
import fullscreenReducer from './reducers/fullscreenReducer';
import loginReducer from './reducers/loginReducer';
import languageReducer from './reducers/languageReducer';

const rootReducer = combineReducers({
  apiError: apiErrorReducer,
  ontology: ontologyReducer,
  fullscreenStatus: fullscreenReducer,
  login: loginReducer,
  languages: languageReducer,
});

const store = createStore(rootReducer);

export type RootState = ReturnType<typeof rootReducer>;

export default store;
