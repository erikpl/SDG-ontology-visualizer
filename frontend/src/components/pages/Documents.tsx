import { Flex, Heading, Stack, Text, Image } from '@chakra-ui/react';
import React from 'react';
import { useSelector } from 'react-redux';
import { useHistory } from 'react-router-dom';
import useTranslation from '../../hooks/translations';
import { RootState } from '../../state/store';
import { MotionBox } from '../../types/react/componentTypes';
import DocumentList from '../molecules/DocumentList';

const Documents: React.FC = () => {
  const history = useHistory();
  const selectedSDG = useSelector((state: RootState) => state.ontology.selectedSDG);
  const selectedSubgoal = useSelector((state: RootState) => state.ontology.selectedSubGoal);
  const translations = useTranslation(); 

  return (
    <Stack spacing="10">
      <Flex
        align="left"
        justify="left"
        justifyContent="space-between"
        spacing="10"
        bg="cyan.700"
        padding="50px"
      >
        <MotionBox
          p={0}
          whileHover={{ scale: 1.05 }}
          _hover={{
            cursor: 'pointer',
          }}
          onClick={() => history.push('/ontology')}
        >
          <Image
            src={translations.getString('icon_'.concat((selectedSDG!.instancesOf.slice(selectedSDG!.instancesOf.indexOf('B') + 1))))} 
            borderRadius="lg"
            overflow="hidden"
            alt={translations.getString(selectedSDG!.instancesOf.slice(selectedSDG!.instancesOf.indexOf('B') + 1))}
            boxSize="200"
            marginRight="5"
            object-fit="cover"
          /> 
        </MotionBox>
        <Stack spacing="4" width="90%" marginLeft='20px'>
          <Heading size="lg" color="white">
            {translations.getString('Target').concat(' ')}
            {selectedSubgoal?.SubjectLabel}
          </Heading>
          <Text fontSize="md" color="white"> 
            {translations.getString('desc'.concat(selectedSDG!.instancesOf.slice(selectedSDG!.instancesOf.indexOf('B') + 1).toString()))} 
          </Text>
        </Stack>
      </Flex>
      <Stack>
        <DocumentList />
      </Stack>
    </Stack>
  );
};
export default Documents;


