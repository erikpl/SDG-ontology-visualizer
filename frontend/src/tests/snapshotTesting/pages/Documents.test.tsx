import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../../../state/store';
import LanguageContextProvider from '../../../contexts/LanguageContextProvider';

import Documents from '../../../components/pages/Documents';

it('renders with no items', () => {
  const tree = renderer
    .create(
      <LanguageContextProvider>
        <Provider store={store}>
          <Documents />
        </Provider>
      </LanguageContextProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
