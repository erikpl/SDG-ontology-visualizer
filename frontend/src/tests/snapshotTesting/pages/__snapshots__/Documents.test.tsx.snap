// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with no items 1`] = `
<div
  className="chakra-stack css-efzghl"
>
  <div
    className="css-1uebyve"
    spacing="10"
  >
    <div
      className="css-1sjj4m4"
      onClick={[Function]}
      style={Object {}}
    >
      <img
        className="chakra-image__placeholder css-1uui2t9"
        object-fit="cover"
      />
    </div>
    <div
      className="chakra-stack css-1mlp23a"
    >
      <h2
        className="chakra-heading css-kbn7if"
      >
        <span>
          Delmål 
        </span>
      </h2>
      <p
        className="chakra-text css-1bqavjz"
      />
    </div>
  </div>
  <div
    className="chakra-stack css-eao00i"
  >
    <div
      align="center"
      className="css-tthxno"
    >
      <h2
        className="chakra-heading css-1qtlgnn"
      >
        Laster dokumenter...
        <div
          className="chakra-spinner css-1y7joxr"
        >
          <span
            className="css-f8n5zr"
          >
            Loading...
          </span>
        </div>
      </h2>
    </div>
  </div>
</div>
`;
